utils/plugin-space-access.js

```diff
@@ -0,0 +1,142 @@
+import fp from "fastify-plugin";
+import {
+  S3,
+  ListObjectsV2Command,
+  HeadObjectCommand,
+} from "@aws-sdk/client-s3";
+
+class PluginSpaceAccessError extends Error {
+  constructor(message, options) {
+    super(message, options);
+  }
+}
+
+class SpaceAccess {
+  constructor(fastify, options) {
+    const {
+      log,
+      hydrateCache,
+      endpoint,
+      region,
+      accessKeyId,
+      secretAccessKey,
+    } = options;
+
+    this.fastify = fastify;
```

Comment:
ℹ️ Not required anymore with caching operations having been moved out of the plugin